extension/js/common/core/pgp.ts

```diff
@@ -246,6 +246,10 @@ export class Pgp {
       const { keys: [key] } = await openpgp.key.readArmored(armoredKey);
       return key;
     },
+    readBinary: async (binaryKey: Uint8Array) => {
+      const { keys: [key] } = await openpgp.key.read(binaryKey);
```

Comment:
I really like that you followed the style above! The trouble with this particular usecase is that we should import all public keys in the file, and there may be many. So returning one file is not so useful, and returning many is inconsistent with `Pgp.key.read`.

`Pgp.key.read` is used to read internal already-stored keys which are guaranteed to be one key per armored block, but keys imported from external sources may contain several, so in this case I would recommend to remove this method and use the code directly.

In general, adding an appropriate method, similar to other methods around, would be the right thing to do. In this particular case, the intended usecase does not match that well.

Maybe we should actually rename `.read` to `.readOne` to make it really clear, but that's for another day.

Reply:
Okey dokey. So I should use `openpgp.read()` directly?

 I can do that, but can you advise how you want me to handle after obtaining the array of keys. Currently I am writing the single returned key into the pubkey input text box and the user will then press the `Show Contents` button. Now assuming the keyring has multiple keys should these be handled in another way. I assume the `Ui.replaceRenderableMsgBlocks` will not work for a concatenated list of key blocks.

Ive been looking over things again, I am not quite clear on how the contact/PGP key is being stored.

Reply:
Sorry I couldn't respond earlier. As it is now it looks better (seems you have removed it and managed to answer your questions :) ).